### DISCUSSION TITLE
Unable to start the server: Specified port is already in use

### DISCUSSION THREAD

**Discussion by sayang14** [minio/minio#15184] [Q&A] [answered]
Hi!, I am trying to run minio server inside a docker container(Linux Ubuntu 22.04 LTS) using 
$ minio server /data and getting this error :

WARNING: MINIO_ACCESS_KEY and MINIO_SECRET_KEY are deprecated.
         Please use MINIO_ROOT_USER and MINIO_ROOT_PASSWORD
ERROR Unable to start the server: Specified port is already in use
      > Please ensure no other program uses the same address/port

Please help me to solve this error. I have just started to learn minio. Thanks!

**Comment by ravindk89** [comment, marked as answer]
The error indicates exactly what it says - something is blocking access to Port 9000, which is the default MinIO Server port.

You can take a look at our [guidance](https://docs.min.io/minio/baremetal/installation/deploy-minio-single-node-single-drive.html#deploy-containerized-single-node-single-drive-minio) on running MinIO as a Single-Node Single-Drive container. 

You can also provide us the command used to run MinIO in a container. 

Without this information we won't be able to help much.

**Reply by sayang14** [reply]
Whenever I am writing
 [root@a7dcb2508242 /]# minio server --address 127.0.0.1:9001 /data

WARNING: MINIO_ACCESS_KEY and MINIO_SECRET_KEY are deprecated.
         Please use MINIO_ROOT_USER and MINIO_ROOT_PASSWORD
Automatically configured API requests per node based on available memory on the system: 224
Finished loading IAM sub-system (took 0.0s of 0.0s to load data).
Status:         1 Online, 0 Offline. 
API: https://127.0.0.1:9001 
RootUser: admin 
RootPass: password 

Console: https://172.20.0.2:37257 https://127.0.0.1:37257 
RootUser: admin 
RootPass: password 

Command-line: https://docs.min.io/docs/minio-client-quickstart-guide
   $ mc alias set myminio https://127.0.0.1:9001 admin password

Documentation: https://docs.min.io

WARNING: Console endpoint is listening on a dynamic port (37257), please use --console-address ":PORT" to choose a static port.


But the above command i.e minio server /data it's not working

**Reply by harshavardhana** [reply]
Yeah @sayang14 port 9000 is not available so use a separate port just like you did. Make sense?

**Reply by ravindk89** [reply]
@sayang14 we have no control over what applications are using existing ports. The above command also has nothing to do with containerization.

A few things to try:

- Run `ps ax | grep minio` and see if you are running MinIO in the background somehow
- Run `sudo lsof -i -P -n | grep LISTEN` and see whether a specific application is already using port 9000

MinIO must have exclusive access to port 9000 (the default port) to run. The fact that specifying `--address :9001` worked indicates something else is running on the default port `9000`.

It is up to you at this point - you can either identify and reconfigure the application using port 9000, or use port 9001. It makes no difference to MinIO, this is just the listen address for the S3 API port.

Just specify `--address :9001` - what you have above is limited MinIO to only localhost adapter, which is unlikely to be what you want.

**Reply by mars-json** [reply]
I am running Minio on stock Ubuntu system, without container.
```
$ minio server ~/minio --console-address :9090
ERROR Unable to start the server: Specified port is already in use
      > Please ensure no other program uses the same address/port
```
This happens after I launched Minio for the first time (in console), and then stopped.
```
$ minio server ~/minio --console-address :9090
Formatting 1st pool, 1 set(s), 1 drives per set.
Automatically configured API requests per node based on available memory on the system: 23
Finished loading IAM sub-system (took 0.0s of 0.0s to load data).
Status:         1 Online, 0 Offline. 
API: http://10.10.20.12:9000  http://10.115.3.10:9000  http://127.0.0.1:9000     
RootUser: minioadmin 
RootPass: minioadmin 

Console: http://10.10.20.12:9090 http://10.115.3.10:9090 http://127.0.0.1:9090   
RootUser: minioadmin 
RootPass: minioadmin 

Command-line: https://docs.min.io/docs/minio-client-quickstart-guide
   $ mc alias set myminio http://10.10.20.12:9000 minioadmin minioadmin

Documentation: https://docs.min.io
^Z
[4]+  Stopped                 minio server ~/minio --console-address :9090
```
It seens Minio server was not stopped, despite information that it is stopped in console.

```
 ps ax | grep minio 
1156772 pts/4    T      0:00 sudo dpkg -i minio.deb
1157376 pts/4    Tl     0:00 minio server /home/vadim/minio --console-address :9090
```
I have installed minio version RELEASE.2022-06-25T15-50-16Z
```
$ minio -v
minio version RELEASE.2022-06-25T15-50-16Z (commit-id=bd099f5e71d0ea511846372869bfcb280a5da2f6)
Runtime: go1.18.3 linux/amd64
Copyright (c) 2015-2022 MinIO, Inc.
```

**Comment by mars-json** [comment]
Started Minio server on port **9001** in console, stopped it, and then tried to start it again.
> ERROR Unable to start the server: Specified port is already in use

It seems Minio server does not free specified port when it is stopped.

```
$ minio server ~/minio --address :9001 --console-address :9091
Automatically configured API requests per node based on available memory on the system: 19
Finished loading IAM sub-system (took 0.0s of 0.0s to load data).
Status:         1 Online, 0 Offline. 
API: http://10.10.20.12:9001  http://10.115.3.10:9001  http://127.0.0.1:9001     
RootUser: minioadmin 
RootPass: minioadmin 

Console: http://10.10.20.12:9091 http://10.115.3.10:9091 http://127.0.0.1:9091   
RootUser: minioadmin 
RootPass: minioadmin 

Command-line: https://docs.min.io/docs/minio-client-quickstart-guide
   $ mc alias set myminio http://10.10.20.12:9001 minioadmin minioadmin

Documentation: https://docs.min.io
^Z
[7]+  Stopped                 minio server ~/minio --address :9001 --console-address :9091

$ minio server ~/minio --address :9001 --console-address :9091
ERROR Unable to start the server: Specified port is already in use
      > Please ensure no other program uses the same address/port
```

**Reply by klauspost** [reply]
> ctrl z is used to pause the process. It will not terminate your program, it will keep your program in background. You can restart your program from that point where you used ctrl z.

> This will generally bring you back to your shell, from which you can generally enter the command bg to move the just-suspended process to the background (letting it continue to run) or fg to bring it back to the foreground.

**Reply by mars-json** [reply]
> 

Frankly, this is not very clear. And I do not see in documentation of MinIO indication how server should be stopped. Probably, docs need improvement. 

You are correct about complete termination of running server:
pressing Ctrl+C terminates server. 
`^C Exiting on signal: INTERRUPT`

And I can restart it again later.

**Reply by klauspost** [reply]
@mars-json This is not minio specific. This is your operating system that operates this way.

**Comment by m0x61h0x64i** [comment]
i have same problem i get the following error when i run the following command

command:
```
MINIO_ROOT_USER=admin MINIO_ROOT_PASSWORD=password minio server /mnt/data --console-address ":666"
```
error:
```
ERROR Unable to start the server: Specified port is already in use
      > Please ensure no other program uses the same address/port
```

i have tried different ports but they also dont work, what should i do ?

**Reply by vadmeste** [reply]
@m0x61h0x64i your 9000 port is probably in use, you can modify it using this flag '--address :9310' (as an example)

**Reply by m0x61h0x64i** [reply]
@vadmeste, thanks!

**Comment by jpainam** [comment]
Hi. 
this problem is still live. How to repoduce. 
1. Start minio.
2. Stop it or close terminal. 
3. Start minio again.

And you will get the error. 
What’s the best way to free the port on minio close?

**Reply by venkatrv** [reply]
I am also facing the same issue:
minio server --console-address "127.0.0.1:9009" C:\MinIOData
FATAL Unable to start the server: listen tcp :9000: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted.

Even I am saying port number as 9009, it comes back with error that, port no: 9000 can't be used !!

**Reply by klauspost** [reply]
There are 2 ports. You are just changing the web port, not the S3 API port.

**Reply by venkatrv** [reply]
Thanks, understood !! How to change the S3 API port?
